src/main/java/org/openrewrite/java/testing/junit5/AssertThrowsOnLastStatement.java

```diff
@@ -113,29 +112,103 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration methodDecl
+                            type = e.getType().toString();
+                        } else if (e.getType() != null && TypeUtils.asClass(e.getType()) != null) {
+                            type = TypeUtils.asClass(e.getType()).getClassName();
+                            maybeAddImport(TypeUtils.asFullyQualified(e.getType()).getFullyQualifiedName(), false);
+                        }
+
+                        J.VariableDeclarations varDecl = new J.VariableDeclarations(Tree.randomId(),
+                                Space.EMPTY,
+                                Markers.EMPTY,
+                                Collections.emptyList(),
+                                Collections.emptyList(),
+                                new J.Identifier(Tree.randomId(),
+                                        Space.EMPTY,
+                                        Markers.EMPTY,
+                                        Collections.emptyList(),
+                                        type.toString(),
+                                        e.getType(),
+                                        null),
+                                null,
+                                Collections.emptyList(),
+                                Arrays.asList(JRightPadded.build(new J.VariableDeclarations.NamedVariable(Tree.randomId(),
+                                        Space.SINGLE_SPACE,
+                                        Markers.EMPTY,
+                                        new J.Identifier(Tree.randomId(),
+                                                Space.EMPTY,
+                                                Markers.EMPTY,
+                                                Collections.emptyList(),
+                                                getVariableName(e),
+                                                e.getType(),
+                                                new JavaType.Variable(null, 0, getVariableName(e), null, e.getType(), null)),
+                                        Collections.emptyList(),
+                                        JLeftPadded.build((Expression)e.withPrefix(Space.SINGLE_SPACE)).withBefore(Space.SINGLE_SPACE),
+                                        new JavaType.Variable(null, 0, getVariableName(e), null, e.getType(), null)))));
+
+                        J.Identifier name = varDecl.getVariables().get(0).getName();
+                        statements.add(varDecl.withPrefix(methodStatement.getPrefix().withComments(emptyList())));
+                        lambdaArguments.add(name);
+                    }
+                    lambdaStatement = mi.withArguments(lambdaArguments);
+                }
+                return lambdaStatement;
+            }
+
+            private String getVariableName(Expression e) {
+                if (e instanceof J.MethodInvocation) {
+                    String name = ((J.MethodInvocation) e).getSimpleName();
+                    name = name.replaceAll("^get", "");
```

Comment:
Should we also check for the `is` prefix of a boolean or will we in this case never have a boolean anyway?